src/lkl/setup.c

```diff
@@ -1355,13 +1353,14 @@ static void init_enclave_clock()
 
     SGXLKL_VERBOSE("Setting enclave realtime clock\n");
 
-    if (oe_is_within_enclave(shm->timer_dev_mem, sizeof(struct timer_dev)))
+    struct timer_dev* t = shm->timer_dev_mem;
+
+    if (!oe_is_outside_enclave(t, sizeof(struct timer_dev)))
```

Comment:
The check can be done when initalizing sgxlkl_enclave_state.shared_memory

Reply:
I wasn't sure so went with the conservative solution. I think you're right though, it's safe to check that during initialization.